compiler-core/src/ast.rs

```diff
@@ -386,9 +386,10 @@ impl<T, E> Function<T, E> {
 pub type UntypedImport = Import<()>;
 
 #[derive(Debug, Clone, PartialEq, Eq)]
-pub struct ImportName {
-    pub location: SrcSpan,
-    pub name: SmolStr,
+pub enum ImportName {
+    Alias(SrcSpan, SmolStr),
+    Original(SrcSpan, SmolStr),
+    Discarded(SrcSpan, SmolStr),
```

Comment:
The `Original` variant here makes it possible for the "original" name to not match the actual name of the module, which should be impossible. Please use `AssignName` instead.

Reply:
There must be no duplication of state in any of the data structures that could result in invalid states being representable. It must use `AssignName`, and the other import struct fields can be checked to determine other facts about the import.

Reply:
I think this one is resolved too!